src/devices/OpenHardwareMonitor/OpenHardwareMonitor.cs

```diff
@@ -0,0 +1,740 @@
+        /// - For each power sensor, this adds another sensor that integrates power over time and so generated the energy used in W/h or
+        /// more conveniently, Kilowatthours (this is the unit the electricity bill bases on)
+        /// - Gives the heat flux for the primary CPU, using the given CPU die size (or a default value)
+        /// </summary>
+        /// <param name="cpuDieSize">Die size of your CPU, optional. Find your CPU on https://en.wikichip.org/ to find out. Note: This
+        /// value is usually much smaller than the size of the physical CPU.</param>
+        public void EnableDerivedSensors(Area cpuDieSize = default)
+        {
+            if (cpuDieSize.Value == 0)
+            {
+                // Values for some recent intel chips (coffee lake)
+                if (LogicalProcessors <= 4)
+                {
+                    cpuDieSize = Area.FromSquareMillimeters(126);
+                }
+                else
+                {
+                    cpuDieSize = Area.FromSquareMillimeters(149.6);
+                }
+            }
+
+            if (_derivedSensors.Count != 0)
+            {
+                // Already set up
+                return;
+            }
+
+            Sensor cpuPower = null;
+
+            foreach (var sensor in GetSensorList())
+            {
+                if (sensor.SensorType == SensorType.Power)
+                {
+                    // Energy usage (integration of power over time)
+                    var managementInstance = new EnergyManagementInstance();
+                    Sensor newSensor = new Sensor(managementInstance, sensor.Name + " Energy", sensor.Identifier + "/energy", sensor.Identifier, SensorType.Energy);
+                    newSensor.Job = StartMonitoring(sensor, TimeSpan.FromMilliseconds(500), (s, value, timeSinceUpdate) =>
+                    {
+                        double previousEnergy = managementInstance.Value;
+                        // Value is in watts, so increment is in watts-hours, which is an unit we can later convert from
+                        double increment = value.Value * timeSinceUpdate.TotalHours;
+                        double newEnergy = previousEnergy + increment;
+                        managementInstance.Value = newEnergy;
+                    });
+                    _derivedSensors.Add(newSensor);
+                }
+
+                // For CPU package, calculate heat flux
+                if (sensor.SensorType == SensorType.Power && sensor.Name.IndexOf("CPU Package", StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    var managementInstance = new EnergyManagementInstance();
+                    var newSensor = new Sensor(managementInstance, sensor.Name + " HeatFlux", sensor.Identifier + "/heatflux", sensor.Identifier, SensorType.HeatFlux);
+                    newSensor.Job = StartMonitoring(sensor, TimeSpan.FromMilliseconds(500), (s, value, timeSinceUpdate) =>
```

Comment:
Same, any reason why 500 ms? Can this be adjusted? Having shortest period would give more precise value if I understand correctly the logic

Reply:
That's true, the shorter the period is, the better the integration works. But relying on the user to know what an integral is, might be a bit demanding, too. I'll think of something.

Reply:
So maybe a default TimeSpan in the function with this default time? Anyone else who'd like something more precise would be able to?

Reply:
Added optional argument